src/BloomBrowserUI/bookEdit/bloomField/bloomField.ts

```diff
@@ -0,0 +1,254 @@
+        }
+        $(field).prepend('<p></p>');
+    }
+
+    static EnsureEndsWithParagraphElement(field:HTMLElement) {
+        //Enhance: move any errant paragraphs to after the imageContainer
+        if($(field).children().length > 0 && ( $(field).children().last().prop("tagName").toLowerCase() === 'p')) {
+            return;
+        }
+        $(field).append('<p></p>');
+    }
+
+    static ConvertTextNodesToParagraphs(field: HTMLElement){
+        var nodes = field.childNodes;
+        for (var n = 0; n < nodes.length; n++) {
+            var node = nodes[n];
+            if (node.nodeType === 3) {//Node.TEXT_NODE
+                var paragraph = document.createElement('p');
+                if(node.textContent.trim() !== '') {
+                    paragraph.textContent = node.textContent;
+                    node.parentNode.insertBefore(paragraph, node);
+                }
+                node.parentNode.removeChild(node);
+            }
+        }
+    }
+
+    // We expect that once we're in paragraph mode, there will not be any cleanup needed. However, there
+    // are three cases where we have some conversion to do:
+    // 1) when a field is totally empty, we need to actually put in a <p> into the empty field (else their first
+    //      text doesn't get any of the formatting assigned to paragraphs)
+    // 2) when this field was already used by the user, and then later switched to paragraph mode.
+    // 3) corner cases that aren't handled by as-you-edit events. E.g., pressing "ctrl+a DEL"
+    static ModifyForParagraphMode(field:HTMLElement) {
+        BloomField.ConvertTextNodesToParagraphs(field);
+        $(field).find('br').remove();
+
+        // in cases where we are embedding images inside of bloom-editables, the paragraphs actually have to go at the
+        // end, for reason of wrapping. See SHRP C1P4 Pupils Book
+        //if(x.startsWith('<div')){
+        if($(field).find('.bloom-keepFirstInField').length > 0){
+            BloomField.EnsureEndsWithParagraphElement(field);
+            return;
+        }
+
+        else{
+               BloomField.EnsureStartsWithParagraphElement(field);
+        }
+    }
+
+    static RequiresParagraphs(field : HTMLElement) : boolean {
+        return $(field).closest('.bloom-requiresParagraphs').length > 0
+                //this signal used to let the css add this conversion after some SIL-LEAD SHRP books were already typed
+            || ($(field).css('border-top-style') === 'dashed');
+    }
+
+    static HandleFieldFocus(field:HTMLElement) {
+
+       BloomField.MoveCursorToFirstParagraph(field, CursorPosition.start);
+        return;
+
+        if (!BloomField.RequiresParagraphs(field)) {
+            return; // nothing to do
+        }
+        if ($(field).text() === '' && $(field).find("p").length === 0) {
+            //stick in a paragraph, which makes FF do paragraphs instead of BRs.
+            $(field).html('<p class="me - initial">&nbsp;</p>');
```

Comment:
It's not obvious why just making the cursor an insertion point in an empty paragraph won't work. Is that not enough to prevent FF switching back to brs?